### PR TITLE
drivers: spi: rv32m1_lpspi: Fix null tx

### LINKED ISSUE #20589: RV32M1 SPI loopback needs DEBUG_OPTIMIZATIONS
Not enabling DEBUG_OPTIMIZATIONS in the spi_loopback test leads to reset loops in the code execution on RV32M1 / the vega board. 
This was discovered while actually enabling the vegaboard for the spi loopback
https://github.com/zephyrproject-rtos/zephyr/pull/20428

**To Reproduce**
Take https://github.com/zephyrproject-rtos/zephyr/pull/20428 and remove  ```CONFIG_DEBUG_OPTIMIZATIONS=y``` from the board config for the spi loopback test. This will work once, but a reset via the reset button will trigger a loop of aborted and restarted UART output.

> ***** Booting Zephyr OS build zephyr-v2.0.0-1813-gcbc791fdf249 *****
Running test suite test_spi
===================================================================
starting test - testing_spi
I: SPI test on buffers TX/RX 0x2000004c/0x200003e0
I: Start
I: CS control inhibited (no GPIO device)�***** Booting Zephyr OS build zephyr-v2.0.0-1813-gcbc791fdf249 *****
Running test suite test_spi
===================================================================
starting test - testing_spi
I: SPI test on buffers TX/RX 0x2000004c/0x200003e0
I: Start
I: CS control inhibited (no GPIO device)�***** Booting Zephyr OS build zephyr-v2.0.0-1813-gcbc791fdf249 *****
Running test suite test_spi
===================================================================
starting test - testing_spi
I: SPI test on buffers TX/RX 0x2000004c/0x200003e0
I: Start
I: CS control inhibited (no GPIO device)�***** Booting Zephyr OS build zephyr-v2.0.0-1813-gcbc791fdf249 *****
Running test suite test_spi

**Expected behavior**
No crash/restarts

**Impact**
This is limited to the spi_loopback sample on RV32M1 for now

**Environment (please complete the following information):**
Toolchain from open-isa.org
> karsten@katzenklo:~/rv32m1/riscv32-unknown-elf-gcc/bin> ./riscv32-unknown-elf-gcc --version
riscv32-unknown-elf-gcc (GCC) 7.1.1 20170509

**Additional context**
Commenting out or shuffling around some of the tests will eventually start to a stable test execution, in particular the `spi_complete_loop(spi_slow, &spi_cfg_slow) || spi_rx_half_start(spi_slow, &spi_cfg_slow) || 	    spi_rx_half_end(spi_slow, &spi_cfg_slow) || spi_rx_every_4(spi_slow, &spi_cfg_slow) || spi_async_call(spi_slow, &spi_cfg_slow))` before `spi_resource_lock_test` blocks (for spi_slow and spi_fast) but that could be down to anything. Also the IRQ table seems to be generated correctly, all the pointers are valid and sensible

### PR DESCRIPTION
**Pull request by karstenkoenig**
Initialize the dummy data transfer so spi transfer is defined even for
an undefined tx data buffer.
This is part of a fix for #20589, together with PR #26031

### CODE REVIEW

**Comment by karstenkoenig** [comment]
Rebased on master and checked again, still works :-)

**Review by MaureenHelm — APPROVED** [review]
thanks @karstenkoenig

**Comment by karstenkoenig** [comment]
@henrikbrixandersen I added you as you did the i2c driver for the rv32m1 so are familiar with the rv32m1, hope this is ok :-)

File: tests/drivers/spi/spi_loopback/boards/rv32m1_vega_ri5cy.conf [hunk 1]
```diff
@@ -6,4 +6,3 @@
 
 CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_0"
 CONFIG_SPI_LOOPBACK_SLAVE_NUMBER=2
-CONFIG_DEBUG_OPTIMIZATIONS=y
```
Comment (henrikbrixandersen) [inline tests/drivers/spi/spi_loopback/boards/rv32m1_vega_ri5cy.conf:9]: Should this change be part of this PR? If yes, could you please separate it into its own commit with a descriptive commit message?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: tests/drivers/spi/spi_loopback/boards/rv32m1_vega_ri5cy.conf (hunk 1 above)
Comment (karstenkoenig) [reply tests/drivers/spi/spi_loopback/boards/rv32m1_vega_ri5cy.conf]: Done :-)

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by MaureenHelm** [comment]
Please rebase to the latest master. I think you're missing the commit that added the buildkite pipeline, hence the check is failing.

**Comment by karstenkoenig** [comment]
> Please rebase to the latest master. I think you're missing the commit that added the buildkite pipeline, hence the check is failing.

Done, thanks!
